src/main/scala/introprog/Swing.scala

```diff
@@ -37,17 +37,10 @@ object Swing {
   /** Check whether `/proc/version` on this filesystem contains any of the strings in `parts`. 
     * Can be used to detect if we are on WSL instead of "real" linux/ubuntu.
     */
-  private def isInProc(parts: String*): Boolean = {
-    val searchExpression = parts.mkString("|")
-    val cmd = Seq(
-      "grep", 
-      "--ignore-case", "--perl-regexp", "--no-messages", "--silent", 
-      searchExpression, 
-      "/proc/version"
-    )
-    val process = sys.process.Process(cmd)
-    util.Try(process.! == 0).getOrElse(false)
-  }
+  private def isInProc(parts: String*): Boolean =
+    util.Try(parts.map(_.toLowerCase)
+    .exists(IO.loadString("/proc/version").toLowerCase.contains(_)))
```

Comment:
As it is chained I think this lambda is easier to read rather than the far away underscore :
part =>IO.loadString("/proc/version").toLowerCase.contains(part)